frontend/discourse/tests/acceptance/login-from-signup-test.js

```diff
@@ -0,0 +1,88 @@
+import { click, fillIn, visit } from "@ember/test-helpers";
+import { test } from "qunit";
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+
+acceptance("Login Fields - From Signup Form", function () {
+  test("autofills login field with signup email value", async function (assert) {
+    await visit("/");
+    await click("header .sign-up-button");
+    await fillIn("#new-account-email", "test@example.com");
+    await fillIn("#new-account-email", "test@example.com");
+
+    await click(".signup-page-cta__login");
+
+    assert
+      .dom("#login-account-name")
+      .hasValue("test@example.com", "login name is autofilled with email");
+  });
+
+  test("autofills login field with signup username value when no email", async function (assert) {
+    await visit("/");
+    await click("header .sign-up-button");
+    await fillIn("#new-account-username", "testuser");
+
+    await click(".signup-page-cta__login");
+
+    assert
+      .dom("#login-account-name")
+      .hasValue("testuser", "login name is autofilled with username");
+  });
+
+  test("prefers email over username when both are filled", async function (assert) {
```

Comment:
```suggestion
  test("prefers email over username when both email and username fields have been filled", async function (assert) {
```

Tiny suggestion for clarity.